tritium-processor/src/test/resources/com/palantir/tritium/examples/ProxiedDelegateToRunnable.java.generated

```diff
@@ -0,0 +1,73 @@
+package com.palantir.tritium.examples;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.UndeclaredThrowableException;
+import javax.annotation.processing.Generated;
+
+@Generated("com.palantir.tritium.processor.ProxyAnnotationProcessor")
+public final class ProxiedDelegateToRunnable implements DelegateToRunnable {
+    private static final Method EQUALS;
+
+    private static final Method TO_STRING;
+
+    private static final Method RUN;
+
+    static {
+        try {
+            EQUALS = DelegateToRunnable.class.getMethod("equals", Object.class);
+            TO_STRING = DelegateToRunnable.class.getMethod("toString");
+            RUN = DelegateToRunnable.class.getMethod("run");
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private final InvocationHandler handler;
+
+    private ProxiedDelegateToRunnable(InvocationHandler handler) {
+        this.handler = handler;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean equals(Object obj) {
+        try {
+            Object returned = this.handler.invoke(this, EQUALS, new Object[]{obj});
+            return (boolean) returned;
+        } catch (Error | RuntimeException throwable) {
+            throw throwable;
+        } catch (Throwable throwable) {
+            throw new UndeclaredThrowableException(throwable);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public String toString() {
+        try {
+            Object returned = this.handler.invoke(this, TO_STRING, new Object[]{});
+            return (String) returned;
+        } catch (Error | RuntimeException throwable) {
+            throw throwable;
+        } catch (Throwable throwable) {
+            throw new UndeclaredThrowableException(throwable);
+        }
+    }
+
+    @Override
+    public void run() {
+        try {
+            Object returned = this.handler.invoke(this, RUN, new Object[]{});
+            void.class.cast(returned);
+        } catch (Error | RuntimeException throwable) {
+            throw throwable;
+        } catch (Throwable throwable) {
+            throw new UndeclaredThrowableException(throwable);
+        }
+    }
+
+    public static DelegateToRunnable of(InvocationHandler handler) {
+        return new ProxiedDelegateToRunnable(handler);
```

Comment:
Do we want to generate `handler != null` precondition check for the constructor or factory?

Reply:
:+1: I'll use `Objects.requireNonNull`, that way this processor can be used with no runtime dependencies.

I think it will make the most sense to break this out into a separate annotation processor, mostly because the annotation is not needed at runtime, while the tritium processor (overloaded at this point, InvocationEventHandler processor?) requires a dependency on tritium-lib.